### DISCUSSION TITLE
How to build the test cases residing under OpenDDS-3.21\tests\DCPS?

### DISCUSSION THREAD

**Discussion by h3gui** [OpenDDS/OpenDDS#3688] [Q&A] [answered]
The following link https://github.com/objectcomputing/OpenDDS/blob/master/INSTALL.md mentioned that the tests under OpenDDS-3.21\tests\DCPS are not automatically built. They could be built by running configure --tests
However, the command does not run successfully due to the fact that OpenDDS-3.21\tests\googletest is empty (Google Test does not seem to be included in the OpebDDS-3.21 package)

**Comment by iguessthislldo** [comment, marked as answer]
It's included in the git repo as a submodule that has to be initialized if it's not already. It looks like you're using a release archive though, so you'll have to download Google Test manually and put it in `tests/googletest`. One way to do this would be to run:

```
git clone https://github.com/oci-labs/googletest tests/googletest -b v1.8.x
```

**Reply by mitza-oci** [reply]
This would be a good addition to dependencies.md and/or the configure script.

**Reply by iguessthislldo** [reply]
Yes, I agree: https://github.com/objectcomputing/OpenDDS/pull/3691. We could also do the same with RapidJSON.
